src/components/views/dialogs/ManageRestrictedJoinRuleDialog.tsx

```diff
@@ -83,11 +83,20 @@ const ManageRestrictedJoinRuleDialog: React.FC<IProps> = ({ room, selected = [],
     const [query, setQuery] = useState("");
```

Comment:
This really feels like a [`z-confidence-high`](https://github.com/vector-im/element-meta/wiki/Triage-process#triaging-enhancements-draft) type of issue where it's an obvious win and doesn't need to be blocked behind product/design that hasn't looked at it yet.

It's using existing UI and adding results you actually care about using. Nothing controversial there and can always be iterated upon again.

---

The current workaround involves doing manual API requests which is obviously not better. Here is what I wrote in our own spaces guide to get around this:

> ### Updating your room to be accessible from the Element Corp. space and your sub-space
> 
> In order to allow people to join your room without joining the sub-space first, you need to update the join rules to point to the Element Corp. space and your specific sub-space. This is important because when you link a random room, not everyone will be part of your sub-space first and will run into `You do not belong to any of the required rooms/spaces to join this room.` when you try to point them to the room.
> 
> Type **`/devtools`** in the room -> **Explore room state** -> **`m.room.join_rules`** -> **Edit** -> Paste the following content (update `<your_subspace_here>` with your specific sub-space) and **Send**
> 
> ```json
> {
>   "allow": [
>     {
>       "room_id": "<your_subspace_here>",
>       "type": "m.room_membership"
>     },
>     {
>       "room_id": "!xxx:matrix.org",
>       "type": "m.room_membership"
>     }
>   ],
>   "join_rule": "restricted"
> }
> ```
>
> *-- https://gitlab.matrix.org/new-vector/internal/-/wikis/Teams/IT-Support/Spaces*

Reply:
This feature was previously rejected from scope for spaces, and does not meet the criteria for that reason - it needs product review to ensure the scope is re-evaluated appropriately.

Reply:
We tie our hands way too often with this type of reasoning.

Reply:
normally I would agree, however the feature quite literally being rejected in the past makes this a rare case, and one that needs the product team's direct involvement. 

The design request can be foregone with the new process, thankfully, but the product team review is exceedingly important.

Reply:
Given that we run into the condition this change improves ourselves quite commonly, I'd second getting it landed. If anything, this makes the join rule setup more usable than it is today. So while maybe not being perfect, it strikes me as a clear improvement.

@daniellekirkwood any objections to getting this landed?